src/_guides/language/language-tour.md

```diff
@@ -2845,10 +2845,10 @@ class ProfileMark {
 ```
 
 If you want to assign the value of a `final` instance variable
-after the constructor body starts, you can use `late final`,
-[but _be careful_][late-final-ivar].
+after the constructor body starts, you can use one of the following:
 
-[late-final-ivar]: /guides/language/effective-dart/design#avoid-public-late-final-fields-without-initializers
+* Use `late final`, but [_be careful_][late-final-ivar].
+* Use a [factory constructor](#factory-constructors).
```

Comment:
I think a factory constructor may be a better first alternative to suggest as `late final` can result in some tricky to identify errors. What do you think?

```suggestion
* Use a [factory constructor](#factory-constructors).
* Use `late final`, but [_be careful_][late-final-ivar].
```

Reply:
sgtm, but I'd like to get the opinions of @natebosch and/or @munificent

Reply:
+1 on swapping the order of the suggestions.

Can we also inline a brief bit of the reason why these are dangerous?

```
* Use `late final`, but [_be careful_][late-final-ivar], this pattern introduces a setter to the API.
```

Reply:
Thanks, both! I've added a variation on Nate's text:

```
* Use a [factory constructor](#factory-constructors).
* Use `late final`, but [_be careful_][late-final-ivar];
  a `late final` without an initializer adds a setter to the API.
```